tests/unit/test_charm_non_leader_unit.py

```diff
@@ -0,0 +1,224 @@
+#!/usr/bin/env python3
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import unittest
+from unittest.mock import Mock, PropertyMock, call, patch
+
+from ops import testing
+from ops.model import ActiveStatus, BlockedStatus, WaitingStatus
+
+from charm import AlertmanagerConfigurerOperatorCharm
+
+testing.SIMULATE_CAN_CONNECT = True
+
+TEST_MULTITENANT_LABEL = "some_test_label"
+TEST_CONFIG = f"""options:
+  multitenant_label:
+    type: string
+    description: |
+      Alertmanager Configurer has been designed to support multiple tenants. In a multitenant
+      Alertmanager Configurer setup, each alert is first routed on the tenancy label, and then
+      the routing tree is distinct for each tenant.
+    default: {TEST_MULTITENANT_LABEL}
+"""
+with open("./tests/unit/test_config/alertmanager_default.yml", "r") as default_yaml:
+    TEST_ALERTMANAGER_DEFAULT_CONFIG = default_yaml.read()
+TEST_ALERTMANAGER_CONFIG_FILE = "/test/rules/dir/config_file.yml"
+ALERTMANAGER_CLASS = "charm.AlertmanagerConfigurerOperatorCharm"
+
+
+class TestAlertmanagerConfigurerOperatorCharmNonLeader(unittest.TestCase):
+    @patch("charm.KubernetesServicePatch", lambda charm, ports: None)
+    def setUp(self):
+        testing.SIMULATE_CAN_CONNECT = True
+        self.harness = testing.Harness(AlertmanagerConfigurerOperatorCharm, config=TEST_CONFIG)
+        self.addCleanup(self.harness.cleanup)
```

Comment:
You may want to set_leader to false explicitely so that it's obvious

Reply:
Done.